Add skipDockerPush and skipDocker properties

cthiebault opened the pull request:
Add
* `skipDocker` to skip any Docker goal (useful when goals are bound to maven phases)
* `skipDockerPush` to skip image push when `push` goal is bound to `deploy` phase

This PR is following https://github.com/spotify/docker-maven-plugin/issues/203

davidxia commented:
@cthiebault Thanks for your PR. In addition to `PushMojo`, `BuildMojo` and `TagMojo` also call `pushImage()`. Could we add a test for this? See examples in `PushMojoTest`.

cthiebault commented:
No problem, I'll do that tomorrow!

cthiebault commented:
I've added some tests for the skip actions.
I've also added `slf4j-simple` for tests so we don't have `Failed to load class org.slf4j.impl.StaticLoggerBinder` error message anymore:
```
<dependency>
  <groupId>org.slf4j</groupId>
  <artifactId>slf4j-simple</artifactId>
  <version>1.7.19</version>
  <scope>test</scope>
</dependency>
```

cthiebault commented:
Any news on this PR? I'd like to use these skip flags...
Maybe I've changed too many files? Do you prefer I reduce the scope of my commits?

guilhox commented:
Hi! Is this PR still being looked into? We have a team eager to use those flags as well...

guzmo commented:
Same here :)

mattnworb commented on src/test/java/com/spotify/docker/BuildMojoTest.java:
```diff
@@ -53,13 +53,10 @@
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.verify;
+import static org.assertj.core.api.Assertions.*;
+import static org.mockito.Mockito.*;
```
would prefer to not use star imports

mattnworb commented on src/test/java/com/spotify/docker/TestUtils.java:
```diff
@@ -0,0 +1,38 @@
+
+  public static File getPomAndAssertExists(String filename) throws Exception {
+    final File pom = new File(TestUtils.class.getResource(filename).toURI());
+    assertThat(pom).isNotNull().exists();
```
very minor but: `pom` can never be null

mattnworb commented:
LGTM overall, and thanks for the contribution @cthiebault. Looks like there are merge conflicts though. Would be happy to merge if you could address the star import thing above.

cthiebault commented:
I've just applied your comments and fixed the conflicts...

mattnworb commented:
@cthiebault thanks again!

derekbassett commented:
This is SO awesome is there any way we can get a new drop of the plugin?  I have a delivery for some internal software and would prefer not to fork our instance of the plugin if possible.

davidxia commented:
@derekbassett I just released 0.4.12.
